File: reactreka/src/main/java/com/microsoft/office/reactreka/ReactContextHolder.java
package com.microsoft.office.reactreka;

import com.facebook.react.bridge.ReactContext;

import java.lang.ref.WeakReference;

public class ReactContextHolder {
    private final WeakReference<ReactContext> mReactContext;
    public ReactContextHolder(ReactContext reactContext) {
        mReactContext = new WeakReference<ReactContext>(reactContext);
    }

    public ReactContext getReactContext() {
        return mReactContext.get();
    }
}

File: sdxruntime/src/main/java/com/microsoft/office/reacthost/ReactContextHolder.kt
package com.microsoft.office.reacthost

import com.facebook.react.bridge.ReactContext
import com.microsoft.office.plat.annotation.KeepClassAndMembers
import java.lang.ref.WeakReference

@KeepClassAndMembers
class ReactContextHolder(reactContext: ReactContext) {
    private val mReactContext: WeakReference<ReactContext>

    init {
        mReactContext = WeakReference(reactContext)
    }

    val reactContext: ReactContext?
        get() = mReactContext.get()
}